code/web/src/modules/user/EditProfile.js

```diff
@@ -0,0 +1,313 @@
+                  style={{ width: 200, marginTop: '1em' }}
+                />
+              ))}
+            </GridCell>
+            <GridCell style={{ padding: '2em', textAlign: 'center', maxWidth: '45%' }}>
+              <H4 font='primary' style={{ marginBottom: '1.13em' }}>Shipping Address</H4>
+
+              {/* Street address */}
+              <Input
+                type="text"
+                fullWidth={true}
+                placeholder='Street address'
+                required="required"
+                name="street"
+                autoComplete="off"
+                value={this.state.newProfileData.street}
+                onChange={this.onChange}
+                style={{ marginBottom: '1.13em' }}
+              />
+
+              {/* City */}
+              <Input
+                type="text"
+                fullWidth={true}
+                placeholder='City'
+                required="required"
+                name="city"
+                autoComplete="off"
+                value={this.state.newProfileData.city}
+                onChange={this.onChange}
+                style={{ marginBottom: '1.13em' }}
+              />
+
+              {/* State Select will go here */}
+              <Select
+                fullWidth={true}
+                required="required"
+                name="state"
+                value={this.state.newProfileData.state}
+                onChange={this.onChange}
+                style={{ marginBottom: '1.13em'}}
+              >
+                <option value='' style={{ color: '#757575'}}>Please select state </option>
+                {
+                  statesList.map(state => (
+                    <option value={state.code} key={state.code}>{state.name}</option>
+                    ))
+                }
+              </Select>
+
+              {/* Zip code */}
+              <Input
+                type="text"
+                fullWidth={true}
+                placeholder='Zip code'
+                required="required"
+                name="zip"
+                autoComplete="off"
+                value={this.state.newProfileData.zip}
+                onChange={this.onChange}
+                style={{ marginBottom: '1em' }}
+              />
+            </GridCell>
+          </Grid>
+          <Grid justifyCenter={true}>
+            <GridCell style={{ maxWidth: '12.45vw'}}>
+              <Button theme='secondary'>Update Profile</Button>
+            </GridCell>
+          </Grid>
+        </form>
+
+        {/* Profile Info Edit form */}
+
+      </div>     
+    )
+  }
+}
+
+// Component Properties
+EditProfile.propTypes = {
+  upload: PropTypes.func.isRequired,
+  messageShow: PropTypes.func.isRequired,
+  messageHide: PropTypes.func.isRequired,
+}
+
+function editProfileState(state) {
+  return {
+    user: state.user,
+  };
+}
+
+export default withRouter(connect(editProfileState, { logout, messageShow, messageHide, upload })(EditProfile))
+
```

Comment:
I would love a walkthrough of this file at some point!! Looks great from what I understand-- but I think an explanation would enhance my learning significantly